test/constructors.jl

```diff
@@ -337,4 +344,11 @@ end
     end
 end
 
+@testset "Dict constructor corner case" begin
+    @test DataFrame(Dict('a' => 1, true => 2)) == DataFrame("a" => 1, "true" => 2)
```

Comment:
Should we really allow this?

Reply:
This was the original design. Actually I thought that maybe it is best to allow only string or `Symbol` as keys - again. It is last time to change it :). Do we go this way?

Reply:
That sounds safer -- as usual. :-)

Reply:
OK - I will change this then. Thank you!

Reply:
done - now we are as strict as with `Pair` constructors (except for sorting when `Dict` is passed).